script.js

```diff
@@ -1,4 +1,250 @@
+      if (playerNumber == 1) {
+        currentPlayerOneScore = Number(combinedNumber);
+        // playerOneScores.push(currentPlayerOneScore);
+        totalPlayerOneScore = totalPlayerOneScore + currentPlayerOneScore;
+      } else if (playerNumber == 2) {
+        currentPlayerTwoScore = Number(combinedNumber);
+        // playerTwoScores.push(currentPlayerTwoScore);
+        totalPlayerTwoScore = totalPlayerTwoScore + currentPlayerTwoScore;
+        if (currentPlayerOneScore > currentPlayerTwoScore) {
+          winningPlayer = "Player 1";
+        } else if (currentPlayerTwoScore > currentPlayerOneScore) {
+          winningPlayer = "Player 2";
+        }
+      }
+
+      myOutputValue = formatString(
+        "welcome",
+        input,
+        combinedNumber,
+        totalPlayerOneScore,
+        totalPlayerTwoScore
+      );
+      myOutputValue =
+        myOutputValue +
+        formatString(
+          "result",
+          input,
+          combinedNumber,
+          totalPlayerOneScore,
+          totalPlayerTwoScore
+        );
+      myOutputValue =
+        myOutputValue +
+        formatString(
+          "leaderboard",
+          input,
+          combinedNumber,
+          totalPlayerOneScore,
+          totalPlayerTwoScore
+        );
+
+      if (playerNumber == 1) {
+        playerNumber = 2;
+      } else if (playerNumber == 2) {
+        playerNumber = 1;
+      }
+    } else {
+      equals = false;
+
+      myOutputValue = formatString(
+        "welcome",
+        input,
+        combinedNumber,
+        totalPlayerOneScore,
+        totalPlayerTwoScore
+      );
+      myOutputValue =
+        myOutputValue +
+        formatString(
+          "leaderboard",
+          input,
+          combinedNumber,
+          totalPlayerOneScore,
+          totalPlayerTwoScore
+        );
+
+      gameMode = "resultMode";
+    }
+  } else {
+    myOutputValue = formatString(
+      "click submit",
+      input,
+      totalPlayerOneScore,
+      totalPlayerTwoScore
+    );
+  }
+}; // end of rollDiceMode
+
+var resultMode = function (input) {
+  if (input == 1 || input == 2) {
+    if (input == 1) {
+      combinedNumber = "" + randomNumberOne + randomNumberTwo;
+    } else if (input == 2) {
+      combinedNumber = "" + randomNumberTwo + randomNumberOne;
+    }
+    if (playerNumber == 1) {
+      currentPlayerOneScore = Number(combinedNumber);
+      // playerOneScores.push(currentPlayerOneScore);
+      totalPlayerOneScore = totalPlayerOneScore + currentPlayerOneScore;
+    } else if (playerNumber == 2) {
+      currentPlayerTwoScore = Number(combinedNumber);
+      // playerTwoScores.push(currentPlayerTwoScore);
+      totalPlayerTwoScore = totalPlayerTwoScore + currentPlayerTwoScore;
+      if (currentPlayerOneScore > currentPlayerTwoScore) {
+        winningPlayer = "Player 1";
+      } else if (currentPlayerTwoScore > currentPlayerOneScore) {
+        winningPlayer = "Player 2";
+      }
+    }
```

Comment:
This code is repeated from above, you might want to refactor it into a helper function!